web-local/tests/utils/ResourceWatcher.ts

```diff
@@ -0,0 +1,33 @@
+import { waitUntil } from "@rilldata/web-common/lib/waitUtils";
+import type { Page } from "playwright";
+import { updateCodeEditor } from "web-local/tests/utils/commonHelpers";
+
+const ResourceWatcherLogRegex = /^\[(.*)] rill\.runtime\.v1\.(.*)\/(.*)$/;
+
+export class ResourceWatcher {
+  private statuses = new Map<string, string>();
+
+  public constructor(private readonly page: Page) {
+    page.on("console", (e) => {
+      const matches = ResourceWatcherLogRegex.exec(e.text());
+      if (!matches || matches.length < 4) return;
+      const [, status, , name] = matches;
+      this.statuses.set(name, status);
+    });
+  }
+
+  public async waitForResource(name: string) {
+    return waitUntil(() => this.statuses.get(name) === "RECONCILE_STATUS_IDLE");
```

Comment:
Our e2e tests should rely on user-visible indicators, not on internal state

Reply:
True. I am iterating on this. That part is still TODO

Reply:
Actually I am not sure if any user-visible indicators was reliable here. That is why we had `waitForResource` earlier.